### ISSUE TITLE
Trying to parseSQL with combinator 'xor' returns empty rules array

### ISSUE THREAD

**Issue by evakratalov** [react-querybuilder/react-querybuilder#375]
i'm trying to parse expression "(type = '' xor test = '')" with parseSQL and getting result with combinator 'and', and empty rules array. 
{rules: Array(0), combinator: 'and'}
will appreciate your help..

**Comment by jakeboone02** [comment]
Hi @evakratalov. To remain compatible with the widest range of SQL engines, the `parseSQL` function only handles ANSI SQL. `XOR` is not part of that standard.

Theoretically you could modify `parseSQL` to handle `XOR`, but you'd need to make a few changes to this repository that I'd be unlikely to merge since it's not standard. If you want to go it alone, here's some information to get you started:

- It looks like [sql.jison](https://github.com/react-querybuilder/react-querybuilder/blob/main/packages/react-querybuilder/src/utils/parseSQL/sql.jison) already supports detection of `XOR`, so you probably wouldn't have to regenerate [sqlParser.js](https://github.com/react-querybuilder/react-querybuilder/blob/main/packages/react-querybuilder/src/utils/parseSQL/sqlParser.js). If you do need to regenerate it, run
  ```
  yarn workspace react-querybuilder generate-sqlparser
  ```
- You would need to add some attributes to the [SQLXORExpression type in types.ts](https://github.com/react-querybuilder/react-querybuilder/blob/main/packages/react-querybuilder/src/utils/parseSQL/types.ts#L175) before modifying `parseSQL` itself. The attributes would probably be the same as `SQLOrExpression`.
- I'm guessing the `parseSQL` modifications would consist of minor alterations to [the existing code that handles `AND`/`OR`](https://github.com/react-querybuilder/react-querybuilder/blob/66a350a230133bdf6557560f059fd49cf39cc450/packages/react-querybuilder/src/utils/parseSQL/parseSQL.ts#L128-L171) since the grammar/syntax is the same; you probably wouldn't need to copy-paste it.
- Since `parseSQL` outputs `DefaultRuleGroupType` you'd need to modify that type to include `"xor"` as a combinator name, which means adding that to the string union type `DefaultCombinatorName`.
- Finally, the `combinators` prop on your `<QueryBuilder />` component should include the `XOR` element. I would do it like this:
  ```tsx
  export const App = () => (
    <QueryBuilder combinators={[ ...defaultCombinators, { name: 'xor', label: 'XOR' }]} />
  );
  ```

I hope that helps! Let me know if you need any more assistance, and if you disagree with the decision not to include `XOR` in this library, let me know your reasoning and I'll reconsider.

**Comment by jakeboone02** [comment]
...ok the more I think about this the more I think I could implement it with minimal disruption (possibly none) to existing users or breaking changes. As long as `defaultCombinators` didn't change, I'd just have to include a warning in the documentation that if you expect the SQL to contain `XOR` then you'll have to set the `combinators` prop to include "xor" as I showed in the last comment. Hmmm...

**Comment by evakratalov** [comment]
Hi @jakeboone02,
I appreciate your help and the quick response!
I do think that it should be possible to add custom combinators.. and to keep the defaultCombinators as is.
If it possible to add that option to the combinators prop, that will be great..

**Comment by evakratalov** [comment]
In addition, our future requirements is to add 'NOT' combinator - the query will be 'A and not B'.
So I'll appreciate if it possible to add support for custom combinators..

**Comment by jakeboone02** [comment]
For the `combinators` prop, you can use whatever values you want. There's no restriction on custom combinators in the `<QueryBuilder />` component.

The issue is that `parseSQL` only recognizes standard keywords, so it will only _produce_ rule groups with "and"/"or" combinators. I think I'm going to try and add `XOR` to that function and see how it goes. I'll reference this issue in a PR soon.

**Comment by jakeboone02** [comment]
@evakratalov Good news: I just merged #377 which adds `XOR` support to `parseSQL`! It will be available in [v5.0](https://github.com/react-querybuilder/react-querybuilder/milestone/2).

**Comment by evakratalov** [comment]
Hi @jakeboone02 thank you for the help!
